src/framework/theme/modal/modal.service.tsx

```diff
@@ -85,14 +84,14 @@ class ModalServiceType {
   }
 }
 
-export interface ModalComponentCloseProps {
-  [key: string]: any;
-  onRequestClose: () => void;
+export interface ModalPresentingConfig {
+  element: React.ReactElement<ModalPresentingBased>;
+  allowBackdrop: boolean;
+  onBackdropPress: () => void;
 }
 
 export interface ModalPresenting {
-  show(element: React.ReactElement<ModalComponentCloseProps>,
-       closeOnBackDrop: boolean): string;
+  show(config: ModalPresentingConfig): string;
 
   hide(identifier: string): void;
```

Comment:
We also can use this method to return an empty string as a modal id.
```
this.id = ModalService.hide(this.id);
// no need to clean-up
```

Reply:
Done